src/connection_string.ts

```diff
@@ -309,10 +300,58 @@ export function parseOptions(
     Object.entries(options).filter(([, v]) => v != null)
   );
 
+  // Validate options that can only be provided by one of uri or object
+
+  if (urlOptions.has('serverApi')) {
+    throw new MongoParseError(
+      'URI cannot contain `serverApi`, it can only be passed to the client'
+    );
+  }
+
   if (objectOptions.has('loadBalanced')) {
     throw new MongoParseError('loadBalanced is only a valid option in the URI');
   }
 
+  // SRV connection string validations
+
+  const nonZeroSrvMaxHosts =
+    objectOptions.get('srvMaxHosts') > 0 || urlOptions.get('srvMaxHosts') > 0;
+
+  const srvSpecificOptionsSpecified =
+    objectOptions.has('srvMaxHosts') ||
+    urlOptions.has('srvMaxHosts') ||
+    objectOptions.has('srvServiceName') ||
+    urlOptions.has('srvServiceName');
+
+  if (isSRV) {
+    // SRV Record is resolved upon connecting
+    mongoOptions.srvHost = hosts[0];
+    // SRV turns on TLS by default, but users can override and turn it off
+    const noUserSpecifiedTLS = !objectOptions.has('tls') && !urlOptions.has('tls');
+    const noUserSpecifiedSSL = !objectOptions.has('ssl') && !urlOptions.has('ssl');
+    if (noUserSpecifiedTLS && noUserSpecifiedSSL) {
+      objectOptions.set('tls', true);
+      objectOptions.set('ssl', true);
+    }
+  } else {
+    if (srvSpecificOptionsSpecified) {
+      throw new MongoParseError(
+        'Cannot use maxSrvHosts nor srvServiceName with a non-srv connection string'
+      );
+    }
+  }
+
+  if (nonZeroSrvMaxHosts) {
+    if (urlOptions.get('loadBalanced') === true) {
+      throw new MongoParseError('Cannot use srvMaxHosts option with loadBalanced');
```

Comment:
this duplicates the logic in validateLoadBalancedOptions, and it throws a different error than the one thrown by validateLoadBalancedOptions - we probably don't want that